packages/grid/x-data-grid/src/internals/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -167,12 +170,31 @@ export const useGridKeyboardNavigation = (
   );
 
   const handleColumnHeaderKeyDown = React.useCallback<
-    GridEventListener<GridEvents.columnHeaderNavigationKeyDown>
+    GridEventListener<GridEvents.columnHeaderKeyDown>
   >(
     (params, event) => {
-      if (!params.field) {
+      let isFromCustomHeader = false;
+      if (params.colDef.renderHeader) {
```

Comment:
If `columnHeaderTitleContentContainer` is always there we can always ignore events coming from inside it. It's simpler than to also check if `colDef.renderHeader` is available.

Reply:
The idea was to do not search in the HTML tree if the user did not used `renderHeader`. This is an over-optimization (not even sure it optimize something)